File: src/ts/magma/app/compile/type/VariadicType.ts
import { Type } from "../../../../magma/app/compile/type/Type";
import { TypeCompiler } from "../../../../magma/app/TypeCompiler";
export class VariadicType implements Type {
	type: Type;
	constructor (type: Type) {
		this.type = type;
	}
	generateType(): string {
		return TypeCompiler.generateType(this.type) + "[]"/*unknown*/;
	}
	isFunctional(): boolean {
		return false/*unknown*/;
	}
	isVar(): boolean {
		return false/*unknown*/;
	}
	generateBeforeName(): string {
		return "..."/*unknown*/;
	}
	generateSimple(): string {
		return TypeCompiler.generateType(this)/*unknown*/;
	}
	is(type: string): boolean {
		return "variadic".equals(type)/*unknown*/;
	}
}

File: src/java/magma/app/compile/type/VariadicType.java
package magma.app.compile.type;

public record VariadicType(Type type) implements Type {
    @Override
    public String generate() {
        return this.type.generate() + "[]";
    }

    public boolean isFunctional() {
        return false;
    }

    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "...";
    }

    @Override
    public String generateSimple() {
        return this.generate();
    }

    public boolean is(String type) {
        return "variadic".equals(type);
    }
}
